docs/create/advanced/troubleshooting.mdx

```diff
@@ -0,0 +1,254 @@
+---
+sidebar_position: 4
+title: Troubleshooting
+description: Troubleshoot common issues and problems with Archway development
+---
+
+# Troubleshooting Development Issues
+
+Here you'll find instructions for troubleshooting common issues and problems that can occur during Archway development.
+
+- [General errors](#general-errors)
+- [Creating project fails](#project-creation)
+- [Deployment failures](#deployment-failures)
+- [RPC errors](#rpc-errors)
+- [Testing](#testing)
+- [File permissions](#permissions-errors)
+
+
+## General Errors
+
+### Process exited with code 1 (ChildProcess)
+
+There are times during development where you may encounter vague error responses from the developer CLI. This can leave you wondering how to proceed, because the error response itself doesn't explain what part of the command failed. These errors are usually triggered by Node's [Child Process](https://nodejs.org/api/child_process.html) module panicking and failing to complete a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+
+```bash
+Failed to deploy project
+Error: Process exited with code 1
+at ChildProcess.done (/home/user/.nvm/versions/node/v17.9.0/lib/node_modules/@archwayhq/cli/node_modules/promisify-child-process/index.cjs:70:19)
+at ChildProcess.emit (node:events:527:28)
+at maybeClose (node:internal/child_process:1090:16)
+at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
+```
+
+To understand the real source of the error, you'll need to execute your command with some debugging flags enabled. To understand how to enable debugging on the developer CLI, review our [debugging instructions](debug).
+
+### SyntaxError: Invalid or unexpected token
+
+You'll get this error response for all Archway commands, if the NPM package for the developer CLI was installed using a legacy version of [Node.js](https://nodejs.org/). To solve this problem update your Node.js and reinstall the [NPM package](https://www.npmjs.com/package/@archwayhq/cli). For information about which versions of Node.js are supported, review the [developer installation instructions](../getting-started/install/#nodejs-and-npm).
+
+## Project Creation
+
+### Error: no such subcommand
+
+When creating a new project with the `archway new` command, if you encounter the error response saying `error: no such subcommand: run-script` or `error: no such subcommand: generate`, it means you're missing a required [Cargo](https://github.com/rust-lang/cargo) package.
+
+To fix the error, make sure you've installed both `cargo-generate` and `cargo-run-script`.
+
+Example:
+
+```bash
+cargo install cargo-generate --features vendored-openssl
+cargo install cargo-run-script
+```
+
+## Deployment Failures
+
+### CosmWasm Rust optimizer fails
+
+If you're having trouble with the `cosmwasm/rust-optimizer`, you may need to update Cargo.toml at the root of your project with another version of the optimizer.
```

Comment:
should we add the link to `cosmwasm/rust-optimizer` repo here?

Reply:
Noted 👍